tests/ui/iter_overeager_cloned.fixed

```diff
@@ -21,18 +21,22 @@ fn main() {
         .iter()
         .flatten().cloned();
 
-    // Not implemented yet
-    let _ = vec.iter().cloned().filter(|x| x.starts_with('2'));
+    let _ = vec.iter().filter(|&x| x.starts_with('2')).cloned();
+
+    let _ = vec.iter().find(|&x| x == "2").cloned();
+
+    {
+        let f = |x: &String| x.starts_with('2');
+        let _ = vec.iter().filter(|&x| f(x)).cloned();
+        let _ = vec.iter().find(|&x| f(x)).cloned();
+    }
```

Comment:
Are we sure dererencing the closure argument wouldn't break logic? What about the argument being a tuple? Could you also add a test with a `move`?

Reply:
This could also be an issue if the dereferenced type doesn't implement some trait: https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=91cf6c202658738b0fba1b53c5510d01

Also, if it doesn't implement `Copy` this won't work if it's `&T` -> `T`

Reply:
@lengyijun Could you add a test for this?